### PR TITLE
runtime: clh: disable virtiofs DAX when FS cache size is 0

### LINKED ISSUE #1100: cloud-hypervisor: disable DAX if FS cache size is 0
Disable DAX when it's not required, for example when the FS cache size is 0

### PR DESCRIPTION
**Pull request by devimc**
Guest consumes 120Mb more of memory when DAX is enabled and the default
FS cache size (8G) is used. Disable dax when it is not required
reducing guest's memory footprint.

Without this patch:

```
7fdea4000000-7fdee4000000 rw-s 18850589 /memfd:ch_ram (deleted)
Size:            1048576 kB
KernelPageSize:        4 kB
MMUPageSize:           4 kB
Rss:              187876 kB
```

With this patch:

```
7fa970000000-7fa9b0000000 rw-s 612001  /memfd:ch_ram (deleted)
Size:            1048576 kB
KernelPageSize:        4 kB
MMUPageSize:           4 kB
Rss:               57308 kB
Pss:               56722 kB
```

fixes #1100

Signed-off-by: Julio Montes <julio.montes@intel.com>

-------

cc @likebreath

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, @devimc! Thanks!

File: src/runtime/virtcontainers/clh.go [hunk 1]
```diff
@@ -1217,22 +1217,28 @@ func (clh *cloudHypervisor) addVolume(volume types.Volume) error {
...
+	cacheSize := int64(0)
+	// numQueues and queueSize are required, let's use the
+	// default values defined by cloud-hypervisor
+	numQueues := int32(1)
+	queueSize := int32(1024)
 
+	if clh.config.VirtioFSCache == virtioFsCacheAlways {
+		dax = true
+		cacheSize = int64(clh.config.VirtioFSCacheSize << 20)
+	}
```
Comment (likebreath) [inline src/runtime/virtcontainers/clh.go:1231]: I think we got a conflict here, when we have both `VirtioFSCacheSize = 0` and `virtioFsCacheAlways`. This is also a bug in the original implementation, but it was `working` as `VirtioFSCacheSize = 0` will be omitted when the struct is serialized to json stream (and clh will use its default `VirtioFSCacheSize` value that is 8G.).

My understanding is `VirtioFSCacheSize` got higher priority than `VirtioFSCache` in our kata configuration file, 
given the way how we disable DAX in kata w/ qemu is merely setting `VirtioFSCacheSize = 0` (regardless of the cache mode). If my understanding is right, I'd suggest to remove this block, and always assign `cacheSize =: int64(clh.config.VirtioFSCacheSize << 20)` .

**Review by likebreath — CHANGES_REQUESTED** [review]
@devimc Thanks again for finding and fixing the issue here. The data looks great. Besides one comments above, I think you will also need to update the version of cloud-hypervisor to get generate the the new OpenApi client code before your patch will work (as you found, `dax=false` will be ignored by the current client code).

**Comment by devimc** [comment]
@likebreath fixed, please take a look, thanks

**Review by likebreath — APPROVED** [review]

**Pull request merged**